### PR TITLE
Fix Avahi

### LINKED ISSUE #429: Subdomains don't work on multisite
**Issue**
I've setup chassis with the following `chassis.local.yaml` file (below).

1. I create a subdomain ex `test.unison-branches.local`
2. Go to URL `test.unison-branches.local` and get 
```
This site can’t be reached

test.unison-branches.local’s server DNS address could not be found.
Search Google for test unison branches local
ERR_NAME_NOT_RESOLVED
```
3. `vagrant provision` doesn't solve the issue
4. following `vagrant reload` doesn't solve the issue

I've tried adding `test.unison-branches.local` to the `chassis.local.yaml` file. Then:
1. I run `vagrant provision` - the `test.unison-branches.local` URL seems to work.
2. If I run a following `vagrant reload` command, it's back to not working.

As I understand from docs: http://docs.chassis.io/en/latest/config/#multisite Specifying the following in `yaml` file
```
hosts:
 - unison-branches.local
multisite: subdomains
```
should make sure that creating and using new subdomains should just work - without needing to add that new subdomain to `yaml` file and the following `vagrant provision`.

**My setup**
1. What operating system do you use? => **macOS Sierra 10.12.6**

2. What version of Vagrant are you running? => **Vagrant 1.9.2**

3. Are you using VirtualBox or VMWare and which version are you using? => **VM 5.1.14** 

4. Do you have a custom YAML file? If so, what does it contain?
```
hosts:
 - unison-branches.local
multisite: subdomains
extensions:
 - Chassis/phpcs
 - Chassis/Xdebug
 - Chassis/db-backup
paths:
 # Path to your project directory relative the Chassis.
 base: unison-branches
 wp: wordpress
 content: content
database:
 name: wordpress
 user: wordpress
 password: vagrantpassword
 prefix: wp_
tester_db:
 name: tests
 user: wordpress
 password: vagrantpassword
 prefix: test_
```
5. Do you have any Chassis extensions installed? If so, what are they?
As above in the yaml file

### LINKED ISSUE #472: Multisite network admin not available
I've run into an issue a few times where the network admin screen is not available, as if the admin user doesn't have the proper capabilities. The menu for visiting the network admin is not available from the admin menu (see screenshot to follow)

screen shot 2017-11-28 at 4 43 45 pm

If I visit the network admin URL directly, I see a `Sorry, you are not allowed to access this page.` error.

So far, I haven't been able to track down the steps to reproduce this issue, and others sharing the same config don't seem to be experiencing this problem.

1. What operating system do you use? Mac OS 10.13.1

2. What version of Vagrant are you running? 2.0.1

3. Are you using VirtualBox or VMWare and which version are you using? VB 5.1.30 r118389 (Qt5.6.3)

4. Do you have a custom YAML file? If so, what does it contain?

```
paths:
    base: ..
    content: content
    wp: wordpress
hosts:
    - main.local
    - site1.main.local
    - site2.main.local
multisite: subdomains

# Define database prefix, since we have tests as well
database:
    name: wordpress
    user: wordpress
    password: vagrantpassword
    prefix: wp_
# Define the extensions we want
extensions:
    - chassis/xdebug
    - chassis/tester
    - chassis/sequelpro
    - chassis/cavalcade
    - chassis/mailhog
    - chassis/memcache
    - chassis/phpcs
```

5. Do you have any Chassis extensions installed? If so, what are they? (see above)

### LINKED ISSUE #475: chassis-hosts failing with D-Bus exception
```
dbus.exceptions.DBusException: org.freedesktop.Avahi.NotSupportedError: Not supported
```

(May be related to two-level deep subdomain, i.e. `foo.bar.vagrant.local`)

### LINKED ISSUE #521: Subdomains Hosts Fail Eventually
I've got subdomains working on a vagrant up, but at some point the hosts mapping stops working. I haven't been able to track down when or why yet, but without fail, the subdomains will stop working after some period of time and use. The main site is still accessible. Also, `/etc/chassis-hosts/conf.d/subdomains` still has all of my subdomains registered in the file.

Running `vagrant reload --provision` will fix them for a short period of time, but `vagrant provision` does not.

1. What operating system do you use?
Mac OSX

2. What version of Vagrant are you running?
vagrant 2.0.3

3. Are you using VirtualBox or VMWare and which version are you using?
Virtualbox 5.2.8

4. Do you have a custom YAML file? If so, what does it contain?

```
# Domain names to use for Vagrant
#
# The first name will be used as the machine's name, subsequent names will be
# aliased in your hosts file and in nginx
#
# (When overriding, include vagrant.local if you need it)
hosts:
  - clientname.local

# Should we use multisite?
multisite: subdomains

# Set database values to save the prefix so Chassis doesn't freak out on
# re-provisions after tests.
database:
    name: wordpress
    user: wordpress
    password: vagrantpassword
    prefix: wp_

# Themes
#
# Install a list of themes, and activate the last one of them
themes:
  - my_theme

# Extensions
#
# Install a list of extensions automatically
extensions:
  - Chassis/Tester
  - Chassis/db-backup
  - Chassis/Xdebug

# PHP version
php: 7.2

# Run database saves on vagrant halts
db-backup:
  halt: yes
  suspend: no
```

5. Do you have any Chassis extensions installed? If so, what are they?

### PR DESCRIPTION
**Pull request by BronsonQuick**
I'm trying another Avahi Package that might work better than https://github.com/Chassis/Chassis/pull/688

Fixes #429 
Fixes #521 
Fixes #472 
Fixes #475

### CODE REVIEW

**Comment by BronsonQuick** [comment]
@roborourke @rmccue I couldn't get the Python library I used to fix this working on boot so I ended up finding another library that does work so we've finally got a fix for the most annoying Chassis bug that's been around.

I also have another branch in progress that will automagically create subsites (for both subdomains and subfolders) with WP-CLI so that'll be coming soon too but for now let's get this bad boy reviewed and merged please! 🙇‍♂️

File: puppet/modules/chassis/templates/local-config-extensions.php.erb [hunk 1]
```diff
@@ -15,3 +15,7 @@ if ( file_exists( <%= path %> ) )
 	require_once <%= path %>;
 
 <%- end -%>
+
+<% if @subdomains != false -%>
```
Comment (roborourke) [inline puppet/modules/chassis/templates/local-config-extensions.php.erb]: Should that be `<%- if`?

Comment (BronsonQuick) [reply puppet/modules/chassis/templates/local-config-extensions.php.erb]: Negative Sir. The `-` are for [space trimming](https://puppet.com/docs/puppet/5.3/lang_template_erb.html#space-trimming-1).

File: puppet/modules/chassis/files/chassis-hosts.service [hunk 2]
```diff
@@ -1,13 +1,14 @@
 [Unit]
-Description=Chassis subdomain hosts updater
+Description=Avahi/mDNS CNAME publisher
```
Comment (roborourke) [inline puppet/modules/chassis/files/chassis-hosts.service]: Is this file still needed? Can't see where it's used

Comment (BronsonQuick) [reply puppet/modules/chassis/files/chassis-hosts.service]: Good catch. This should be deleted now that it uses a template! Reckon you could delete it as I won't have my laptop with me for a few weeks? Taa muchly!

**Comment by roborourke** [comment]
Thanks @kucrut! And thanks for your patience @BronsonQuick

**Pull request merged**